pool.go

```diff
@@ -240,9 +248,14 @@ func (p *pool) fetchWith(ctx context.Context, c cid.Cid, with string) (blk block
 	}
 
 	for i := 0; i < len(nodes); i++ {
-		blk, err = p.fetchAndUpdate(ctx, nodes[i], c, i)
+		blk, err = p.fetchAndUpdate(ctx, nodes[i], c, i, transientErrs)
 
 		if err == nil {
+			// downvote all parked failed nodes as some other node was able to give us the required content here
+			for node := range transientErrs {
+				p.downvote(node)
+			}
```

Comment:
how much complexity would it introduce to have this batch of operations occur within one lock acquisition? (i'm okay for this to be a subsequent PR)

Reply:
Hey, have done this. Please take a look.